src/workerd/api/tests/pipe-streams-test.js

```diff
@@ -173,6 +173,691 @@ export const pipeThroughJsToJs = {
+  },
+};
+
+// Test simple pipeTo from internal readable to JavaScript writable
+export const pipeToInternalToJsSimple = {
+  async test() {
+    const enc = new TextEncoder();
+    const dec = new TextDecoder();
+
+    const { readable, writable } = new IdentityTransformStream();
+
+    const chunks = [];
+    const ws = new WritableStream({
+      write(chunk) {
+        chunks.push(chunk);
+      },
+    });
+
+    const pipe = readable.pipeTo(ws);
+
+    const writer = writable.getWriter();
+    writer.write(enc.encode('hello'));
+    writer.write(enc.encode('there'));
+    writer.close();
+
+    await pipe;
+
+    strictEqual(dec.decode(chunks[0]), 'hello');
+    strictEqual(dec.decode(chunks[1]), 'there');
+
+    ok(!ws.locked);
+    ok(!readable.locked);
+
+    const writer2 = ws.getWriter();
+    await rejects(writer2.write('no'), {
+      message: 'This WritableStream has been closed.',
+    });
+  },
+};
+
+// Test pipeTo error in internal readable aborts JS writable when preventAbort = false
+export const pipeToInternalToJsError = {
+  async test() {
+    const enc = new TextEncoder();
+
+    const { readable, writable } = new IdentityTransformStream();
+
+    const ws = new WritableStream({
+      write(chunk) {},
+    });
+
+    const pipe = readable.pipeTo(ws);
+
+    writable.abort(new Error('boom'));
+
+    await rejects(pipe, { message: 'boom' });
+
+    const writer = ws.getWriter();
+    await rejects(writer.write('hello'), { message: 'boom' });
+  },
+};
+
+// Test pipeTo error in internal readable does not abort JS writable when preventAbort = true
+export const pipeToInternalToJsErrorPrevent = {
+  async test() {
+    const { readable, writable } = new IdentityTransformStream();
+
+    const ws = new WritableStream({
+      write(chunk) {},
+    });
+
+    const pipe = readable.pipeTo(ws, { preventAbort: true });
+
+    writable.abort(new Error('boom'));
+
+    await rejects(pipe, { message: 'boom' });
+
+    ok(!ws.locked);
+
+    const writer = ws.getWriter();
+    await writer.write('hello');
+  },
+};
+
+// Test pipeTo closing internal readable closes JS writable when preventClose = false
+export const pipeToInternalToJsClose = {
+  async test() {
+    const { readable, writable } = new IdentityTransformStream();
+
+    const ws = new WritableStream({});
+
+    const pipe = readable.pipeTo(ws);
+
+    writable.close();
+
+    await scheduler.wait(1);
+
+    const writer = ws.getWriter();
+    await rejects(writer.write('hello'), {
+      message: 'This WritableStream has been closed.',
+    });
+  },
+};
+
+// Test pipeTo closing internal readable does not close JS writable when preventClose = true
+export const pipeToInternalToJsClosePrevent = {
+  async test() {
+    const { readable, writable } = new IdentityTransformStream();
+
+    const ws = new WritableStream({});
+
+    const pipe = readable.pipeTo(ws, { preventClose: true });
+
+    writable.close();
+
+    await scheduler.wait(1);
```

Comment:
Can you add a comment to every scheduler.wait(1) describing the intention behind this change? It's not obvious.